### PR TITLE
Internet domain value object implementation

### PR DESCRIPTION
**Pull request by msadikkoseaupa**

### CODE REVIEW

File: tests/Nox.Types.Tests/Types/InternetDomain/InternetDomainTests.cs [hunk 1]
```diff
@@ -1,11 +1,73 @@
...
+        // Arrange & Act
+        var exception = Assert.Throws<TypeValidationException>(() => _ = InternetDomain.From(invalidDomain));
+
+        // Assert
+        Assert.NotNull(exception);
+        Assert.Equal($"Could not create a Nox InternetDomain type with unsupported value '{invalidDomain}'.", exception.Errors.First().ErrorMessage);
+    }
+    
+    [Fact]
+    public void Nox_InternetDomain_Equality_Tests()
```
Comment (rochar) [inline tests/Nox.Types.Tests/Types/InternetDomain/InternetDomainTests.cs]: IMO starting with Nox_ does not bring additional value, what do you think?

Compare_Same_InternetDomians_WithEqual_IsTrue for the Test name, what do you think? Scenario, Methods, Expected result? 

https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#best-practices

File: tests/Nox.Types.Tests/Types/InternetDomain/InternetDomainTests.cs [hunk 2]
```diff
@@ -1,11 +1,73 @@
...
+        var internetDomain1 = InternetDomain.From("example.com");
+        var internetDomain2 =  InternetDomain.From("example.com");
+
+        // Assert
+        Assert.Equal(internetDomain1, internetDomain2);
+    }
+    
     [Fact]
-    public void When_Create_Should()
+    public void Nox_InternetDomain_NotEqual_Tests()
```
Comment (rochar) [inline tests/Nox.Types.Tests/Types/InternetDomain/InternetDomainTests.cs]: same as previous comment

File: tests/Nox.Types.Tests/Types/InternetDomain/InternetDomainTests.cs [hunk 3]
```diff
@@ -1,11 +1,73 @@
...
     {
+        // Arrange & Act
+        var internetDomain1 = InternetDomain.From("example.com");
+        var internetDomain2 =  InternetDomain.From("example.org");
 
+        // Assert
+        Assert.NotEqual(internetDomain1, internetDomain2);
     }
 }
```
Comment (rochar) [inline tests/Nox.Types.Tests/Types/InternetDomain/InternetDomainTests.cs:75]: I would add test to state that "Http://example.com" is not a valid domain address, so this is clear to everyone

**Review by rochar — CHANGES_REQUESTED** [review]
looks, good, add a few small comments

File: tests/Nox.Types.Tests/Types/InternetDomain/InternetDomainTests.cs (hunk 3 above)
Comment (msadikkoseaupa) [reply tests/Nox.Types.Tests/Types/InternetDomain/InternetDomainTests.cs:75]: Additional tests were added

File: tests/Nox.Types.Tests/Types/InternetDomain/InternetDomainTests.cs (hunk 2 above)
Comment (msadikkoseaupa) [reply tests/Nox.Types.Tests/Types/InternetDomain/InternetDomainTests.cs]: Method name was updated as desired.

File: tests/Nox.Types.Tests/Types/InternetDomain/InternetDomainTests.cs (hunk 1 above)
Comment (msadikkoseaupa) [reply tests/Nox.Types.Tests/Types/InternetDomain/InternetDomainTests.cs]: Agree with you. I updated the method name.

**Review by rochar — DISMISSED** [review]

**Review by jan-schutte — APPROVED** [review]

**Pull request merged**